src/Compilers/Core/Portable/Operations/ControlFlowGraph.cs

```diff
@@ -266,7 +265,7 @@ internal bool TryGetLocalFunctionControlFlowGraph(IMethodSymbol localFunction, C
             }
 
             controlFlowGraph = _lazyLocalFunctionsGraphs[info.ordinal];
-            Debug.Assert(controlFlowGraph.Parent == this);
+            Debug.Assert(controlFlowGraph!.Parent == this);
```

Comment:
💡 If you use a `ref` local above, flow analysis will be able to correctly follow the null state for the array element

```csharp
ref var lazyLocalFunctionGraph = ref _lazyLocalFunctionsGraphs[info.ordinal];
```

Then use `lazyLocalFunctionGraph` everywhere the indexer expression currently is.